lab_5_scrapper/scrapper.py

```diff
@@ -90,75 +214,151 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self.config = config
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        links = article_bs.find_all('a')
+        for link in links:
+            url = link.get('href')
+            if url and url.count('/') == 4 and url[:9] == '/novosti/':
+                url = 'https://www.zebra-tv.ru' + url
+                if url not in self.urls and len(self.urls) < self.config.get_num_articles():
+                    self.urls.append(url)
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        chrome_options = ChromeOptions()
+        chrome_options.add_argument('--start-maximized')
+        if self.config.get_headless_mode():
+            chrome_options.add_argument('--headless=new')
+
+        driver = webdriver.Chrome(options=chrome_options)
+        driver.get(self.config.get_seed_urls()[0])
+
+        scroll_pause_time = 2
+        last_height = driver.execute_script("return document.body.scrollHeight")
+        while len(self.urls) < self.config.get_num_articles():
+            driver.execute_script("window.scrollTo(0,document.body.scrollHeight)")
+            current_html = driver.page_source
+            bs = BeautifulSoup(current_html, 'lxml')
+            self._extract_url(bs)
+
+            time.sleep(scroll_pause_time)
+            new_height = driver.execute_script("return document.body.scrollHeight")
+            if new_height == last_height:
+                break
+            last_height = new_height
+        driver.quit()
 
     def get_search_urls(self) -> list:
         """
         Returns seed_urls param
         """
-        pass
+        return self.config.get_seed_urls()
 
 
 class HTMLParser:
     """
     ArticleParser implementation
     """
 
-    def __init__(self, full_url: str, article_id: int, config: Config) -> None:
+    def __init__(self, article_url: str, article_id: int, config: Config) -> None:
```

Comment:
do not change fuction interface, discard

Reply:
done